src/groupeddataframe/complextransforms.jl

```diff
@@ -30,31 +30,36 @@ function _combine_with_first((first,)::Ref{Any},
     @assert first isa Union{NamedTuple, DataFrameRow, AbstractDataFrame}
     extrude = false
 
+    lgd = length(gd)
     if first isa AbstractDataFrame
         n = 0
         eltys = eltype.(eachcol(first))
     elseif first isa NamedTuple{<:Any, <:Tuple{Vararg{AbstractVector}}}
         n = 0
         eltys = map(eltype, first)
     elseif first isa DataFrameRow
-        n = length(gd)
+        n = lgd
         eltys = [eltype(parent(first)[!, i]) for i in parentcols(index(first))]
     elseif !firstmulticol && first[1] isa Union{AbstractArray{<:Any, 0}, Ref}
         extrude = true
         first = wrap_row(first[1], firstcoltype(firstmulticol))
-        n = length(gd)
+        n = lgd
         eltys = (typeof(first[1]),)
     else # other NamedTuple giving a single row
-        n = length(gd)
+        n = lgd
         eltys = map(typeof, first)
         if any(x -> x <: AbstractVector, eltys)
             throw(ArgumentError("mixing single values and vectors in a named tuple is not allowed"))
         end
     end
     idx = idx_agg === NOTHING_IDX_AGG ? Vector{Int}(undef, n) : idx_agg
+    sizehint!(idx, lgd)
     local initialcols
     let eltys=eltys, n=n # Workaround for julia#15276
-        initialcols = ntuple(i -> Tables.allocatecolumn(eltys[i], n), _ncol(first))
+        initialcols = ntuple(_ncol(first)) do i
+            res = Tables.allocatecolumn(eltys[i], n)
+            return res
+        end
```

Comment:
What's the point of this change? The new form is much more verbose.

Reply:
yes - now it is not needed; I will remove (I wanted to do `sizehint!` but it does not work always)